beta/nncf/tensorflow/batchnorm_adaptation.py

```diff
@@ -25,7 +26,6 @@ def run(self, model) -> None:
 
         :param model: A model for which the algorithm will be applied.
         """
-        # TODO(andrey-churkin): Should be implemented.
 
-        raise NotImplementedError('There is no possibility to start the batch-norm statistics adaptation algorithm '
-                                  'for the TensorFlow backend because it is not implemented.')
+        bn_adaptation_runner = DataLoaderBNAdaptationRunner(model, self._device, self._num_bn_forget_steps)
```

Comment:
What are the advantages of implementing the batchnorm adaptation algorithm via DataLoaderBNAdaptationRunner?

Reply:
Not that strong advantages, but anyway:
- The `model` is not an attribute of `TFBatchnormAdaptationAlgorithmImpl`, it is passed as an argument to the `run` method. It would require attaching the `model` as an attribute to the `TFBatchnormAdaptationAlgorithmImpl` class object or passing the model through almost all methods manually. 
- To be more aligned with `PyTorch`.
- `TFBatchnormAdaptationAlgorithmImpl` and `PTBatchnormAdaptationAlgorithmImpl` are similar - there is an option for a common structure.